server/ansible/roles/pbench-server-activate-create-results-dir-structure/tasks/main.yml

```diff
@@ -51,3 +87,7 @@
     state: directory
   with_items:
     - "002"
+
+- name: Apply new SELinux file context to filesystem
+  command:
+    cmd: "restorecon -irv {{ pbench_dir }}"
```

Comment:
The only problem with this command is for existing installs which lots and lots of data already present.  This will take eons in some settings.

Maybe what we can do is store the top of the hierarchies, and then start some background command to finish the task?

Not sure exactly how to handle this.

Reply:
Yes; retrofitting is going to be a problem.

But it's not like we have a lot of deployed servers that need fixing. Maybe we just individually set the directories the installer creates, and for now rely on manually fixing the rest of the tree? This will all go away when/if we move completely to S3 storage.

Reply:
I can do a non-recursive `restorecon` on the top level directories. We depend on inheritance in any case to set the SELinux context. If we start things right. then it should just work(TM).

Instead of worrying about any potential mislabeling here, we can add the check to the server audit and if that ever detects a problem, we fix it by hand. Assuming that't it's a rare (or never) occurrence, that should be OK. If it happens more often, then we add a relabeling cron job.

WDYT?

Reply:
I got rid of the -r. The auditing and fixing are not addressed here. I'll open an issue.